tvloader/parser2v1/parse_snippet.go

```diff
@@ -34,6 +34,10 @@ func (parser *tvParser2_1) parsePairFromSnippet2_1(tag string, value string) err
 	// tag for creating new package section and going back to parsing Package
 	case "PackageName":
 		parser.st = psPackage2_1
+		// check here whether the last file of the previous package contained the FileSpdxIdentifier
```

Comment:
When parsing Snippets, I don't think you want to do the check here in the `PackageName` tag. Rather, `SnippetSPDXID` is what signals the end of the previous File section and the beginning of the current Snippet section.

Because of that, the section of `parse_snippet.go` that handles the `SnippetSPDXID` tag is where I think this check should go.

Finally, also note the comments earlier about communicating to the user via the error message about which File's filename we're talking about.

Reply:
Also, sorry, I meant to ask -- is there a test corresponding to this code? I don't see any additions made to parse_snippet_test.go.